benchmark/CQRRPT_cond_orth_effect.cc

```diff
@@ -0,0 +1,163 @@
+#include "RandLAPACK.hh"
+#include "rl_blaspp.hh"
+#include "rl_lapackpp.hh"
+
+#include <RandBLAS.hh>
+#include <stdio.h>
+#include <string.h>
+#include <math.h>
+#include <numeric>
+#include <iostream>
+#include <fstream>
+#include <chrono>
+#include <thread>
+#include <fstream>
+
+template <typename T>
+static void
+test_cond_helper_0(int64_t m, 
+                  int64_t n, 
+                  int64_t true_k,
+                  const std::tuple<int, T, bool>& mat_type, 
+                  RandBLAS::base::RNGState<r123::Philox4x32> state) {
+
+    std::vector<T> A(m * n, 0.0);
+    std::vector<T> I_ref(n * n, 0.0);
+
+    // Generate random matrix
+    RandLAPACK::util::gen_mat_type(m, n, A, true_k, state, mat_type);
+    RandLAPACK::util::eye(n, n, I_ref);
+
+    // CHOL QR
+    RandLAPACK::CholQRQ<T> Orth_CholQR(false, false);
+    // Orthonormalize A
+    Orth_CholQR.call(m, n, A);
+
+    T* A_dat = A.data();
+    T* I_ref_dat = I_ref.data();
+
+    // Check orthogonality of Q
+    // Q' * Q  - I = 0
+    blas::gemm(Layout::ColMajor, Op::Trans, Op::NoTrans, n, n, m, 1.0, A_dat, m, A_dat, m, -1.0, I_ref_dat, n);
+    T norm_Q = lapack::lange(lapack::Norm::Fro, n, n, I_ref_dat, n);
+    printf("COND(A^{pre}): %21e\n", std::get<1>(mat_type));
+    printf("FRO NORM OF Q' * Q - I: %e\n\n", norm_Q);
+}
```

Comment:
The orthogonality check requires that $\operatorname{rank}(A) = n$. But it looks like the problem setup code ensures $\operatorname{rank}(A) = \texttt{true}\\_\texttt{k}$, where it's possible to get $\texttt{true}\\_\texttt{k} < n$. Am I missing something here?

Reply:
No, you are corrent, true rank parameter should not be there.